### DISCUSSION TITLE
Node query help

### DISCUSSION THREAD

**Discussion by marcuslindblom** [ravendb/ravendb#17817] [Q&A] [answered]
I would like to combine two queries so I only need one roundtrip to the server. In C# I can use something like Lazy but how can I do this in Node?

This is how I do it today:

```js
const home = await session
  .query({ indexName: 'Content/ByUrl' })
  .whereEquals('url', '/')
  .firstOrNull();


const notes = await session
  .query({ indexName: 'Content/ByUrl' })
  .whereEquals('collection', 'posts')
  .whereEquals('published', true)
  .orderByDescending('publishedDate')
  .selectFields(['heading', 'summary', 'publishedDate', 'url'])
  .take(3)
  .all();
```

**Comment by ayende** [comment, marked as answer]
The same way you would do that in C#, something like:

```
const homeLazy = session
  .query({ indexName: 'Content/ByUrl' })
  .whereEquals('url', '/')
  .lazily();

const notesLazy = session
  .query({ indexName: 'Content/ByUrl' })
  .whereEquals('collection', 'posts')
  .whereEquals('published', true)
  .orderByDescending('publishedDate')
  .selectFields(['heading', 'summary', 'publishedDate', 'url'])
  .take(3)
  .lazily();

const home = (await homeLazily.getValue()).firstOrDefault();
const notes = await notesLazy.getValue();

```

**Reply by marcuslindblom** [reply]
@ayende `firstOrDefault()` does not exist, does it exist in the nightly builds?

**Reply by ayende** [reply]
Sorry, it's from:
http://www.java2s.com/ref/javascript/javascript-array-firstordefault-callback-defaultreturn.html

**Reply by marcuslindblom** [reply]
Does it make sense to do something like this or does't it matter in terms of performance?

```js
const { home, notes } = await Promise.all([homeLazy.getValue(), notesLazy.getValue()]).then(([home, notes]) => {
    return {
      home: home.firstOrDefault(),
      notes: notes
    }
  });
```

**Reply by ayende** [reply]
Not really. Practically speaking, only a single async is going to happen, the first one.
The other one is going to complete syncronously.

**Reply by marcuslindblom** [reply]
Last one, can I get query statistics from this single query? I guess use `.statistics(...` on the first one?

**Reply by ayende** [reply]
Lazy supports statistics, yes. You can have statistics for each of those.
Would only be relevant after the query actually materialized, of course.
